sql/core/src/main/scala/org/apache/spark/sql/execution/adaptive/AQEShuffleReadExec.scala

```diff
@@ -207,47 +208,38 @@ case class AQEShuffleReadExec private(
   }
 
   @transient override lazy val metrics: Map[String, SQLMetric] = {
-    if (shuffleStage.isDefined) {
```

Comment:
can we add `assert(shuffleStage.isDefined)` like what we did in `lazy val shuffleRdd`?